sklearnex/linear_model/ridge.py

```diff
@@ -14,11 +14,416 @@
 # limitations under the License.
 # ===============================================================================
 
-from daal4py.sklearn.linear_model import Ridge
-from onedal._device_offload import support_input_format
+import logging
 
-# Note: `sklearnex.linear_model.Ridge` only has functional
-# sycl GPU support. No GPU device will be offloaded.
-Ridge.fit = support_input_format(queue_param=False)(Ridge.fit)
-Ridge.predict = support_input_format(queue_param=False)(Ridge.predict)
-Ridge.score = support_input_format(queue_param=False)(Ridge.score)
+from daal4py.sklearn._utils import daal_check_version, sklearn_check_version
+
+if daal_check_version((2024, "P", 600)):
+    import numbers
+
+    import numpy as np
+    from scipy.sparse import issparse
+    from sklearn.linear_model import Ridge as _sklearn_Ridge
+    from sklearn.metrics import r2_score
+    from sklearn.utils.validation import check_is_fitted, check_X_y
+
+    from daal4py.sklearn._n_jobs_support import control_n_jobs
+    from daal4py.sklearn.linear_model._ridge import _fit_ridge as daal4py_fit_ridge
+
+    if sklearn_check_version("1.0") and not sklearn_check_version("1.2"):
+        from sklearn.linear_model._base import _deprecate_normalize
+    if sklearn_check_version("1.1") and not sklearn_check_version("1.2"):
+        from sklearn.utils import check_scalar
+
+    from onedal.linear_model import Ridge as onedal_Ridge
+    from onedal.utils import _num_features, _num_samples
+
+    from .._device_offload import dispatch, wrap_output_data
+    from .._utils import PatchingConditionsChain
+
+    if sklearn_check_version("1.6"):
+        from sklearn.utils.validation import validate_data
+    else:
+        validate_data = _sklearn_Ridge._validate_data
+
+    def _is_numeric_scalar(value):

```

Comment:
Still not sure why this was/is a separate function...

Reply:
single point of change for expression applied in different parts of the class

Reply:
resolving as per our conversation @icfaust